File: gaia_pipeline/sor-data-transfer/src/main/scala/domains/Task.scala
package domains

case class ActivateDataLakeSaving(
  isSaving: Boolean
)

case class TaskInput(
  sentence: String,
  project: String,
  groupTask: String,
  task: TaskObject,
  taskId: Option[String],
  scheduleId: Option[String],
  taskConfig: Option[String]
  )

case class TaskObject(
  title: String,
  priority: String,
  status: String,
  startDate: String,
  deadline: String,
  duration: String,
)

File: GAIA/gaia_bot/domain/mapper/task.py
class Task:

    def __init__(self):
        pass
        
    def map_task(self, sentence, project, group_task, title, 
                 description, priority, status, start_date, deadline, 
                 duration, user_id):
        return {
            'sentence': sentence,
            'project': project,
            'group_task': group_task,
            'task': {
                'title': title,
                'description': description,
                'priority': priority,
                'status': status,
                'start_date': start_date,
                'deadline': deadline,
                'duration': duration,
            },
            'user_id': user_id
        }
